File: 1-TwoSum/1-TwoSum.java
// Last updated: 3/6/2026, 9:31:41 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {

        int l = nums.length;
        boolean found = false;
        int[] output;

        
        for(int i = 0; i < l; i++){
            for(int j = 0; j < l; j++){
                if(nums[i] + nums[j] == target && i != j){
                    found = true;
                }
                if(found == true){
                    return new int[]{i, j};
                }
            }
        }
        return new int[]{0,0};
    }
}

File: 1-TwoSum/1-TwoSum.py
# Last updated: 3/27/2026, 1:48:33 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        """
        1. two pass hash is used here, but one pass hash also works here because if i was on index 0 and the valid complement was index 2, i dont need to worry that i didnt find it as ill see my index 0 in my hash when im at index 2
        """

        map = {}

        for i,num in enumerate(nums):
            map[num] = i

        for index, current in enumerate(nums):

            complement = target - current
            if complement in map and index != map[complement]:

                return [index, map[complement]]

        return []
